bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/VoiceManagerImpl.java

```diff
@@ -191,18 +191,24 @@ public void say(String text, String voiceId, String sinkId, PercentType volume)
                 if (audioFormat != null) {
                     AudioStream audioStream = tts.synthesize(text, voice, audioFormat);
 
-                    try {
-                        // get current volume
-                        PercentType oldVolume = audioManager.getVolume(sinkId);
-                        // set notification sound volume
-                        if (volume != null) {
-                            audioManager.setVolume(volume, sinkId);
+                    // get current volume
+                    PercentType oldVolume = audioManager.getVolume(sinkId);
```

Comment:
You should handle volume only if you process the stream.
So I will suggest to move your "volume" stuff inside the if just below.